File: yh_turtle/scripts/turtle_clear.py
#!/usr/bin/python3
#-*- coding:utf-8 -*-

import rospy
from std_srvs.srv import Empty

class TurtleClear:
    def __init__(self):
        super(TurtleClear, self).__init__()
        self.clear_client = rospy.ServiceProxy("clear", Empty)

    def call(self): # 서비스 통신(서버-클라이언트)
        try:
            self.clear_client()
        except rospy.ServiceException as e:
            rospy.logerr(f"Failed : {e}")

if __name__ == "__main__":
    rospy.init_node("turtle_clear") # 노드 이름 초기화
    turtle_clear = TurtleClear() # 클래스(TurtleClear())에 인스터스를 만들어 변수에 넣어줌
    turtle_clear.call()

File: yh_turtle/src/turtle_clear.cpp
// 이미 있는 서비스에 맞게 노드를 만든다.
#include "ros/ros.h"
#include "std_srvs/Empty.h"

int main(int argc, char** argv){
    
    ros::init(argc, argv, "turtle_clear");
    ros::NodeHandle nh;

    // 서비스 클라이언트
    ros::ServiceClient clear_client = nh.serviceClient<std_srvs::Empty>("clear");

    // request보낼 서비스 생성
    std_srvs::Empty srv;

    if(clear_client.call(srv)){ // 통신이 잘 되면 if문
        ROS_INFO("Clear");
    }
    else{ // 통신이 잘 안되면 else
        ROS_ERROR("Failed");
        return 1;
    }

    return 0;
}